go/libraries/doltcore/sqle/enginetest/dolt_server_test.go

```diff
@@ -336,3 +350,23 @@ func newConnection(t *testing.T, serverConfig sqlserver.ServerConfig) (*dbr.Conn
 	sess := conn.NewSession(nil)
 	return conn, sess
 }
+
+func TestDoltServerRunningUnixSocket(t *testing.T) {
+	// with default host = 'localhost', the server will use unix socket located in /tmp/mysql.sock
+	sc, serverConfig := startServerWithDefaultConfigOnly(t)
+	sc.WaitForStart()
+
+	t.Run("connecting mysql works", func(t *testing.T) {
+		conn, sess := newConnection(t, serverConfig)
```

Comment:
It would be good to somehow assert that we are actually using the unix socket. It could be verifying the DSN string contains "unix" or something, or maybe there's a way to query the connections and verify that it's a unix domain socket address or something. 

It might also be a good idea to test that we can connect over both a TCP socket and a unix socket at the same time, since they not mutually exclusive.